tests/unit/test_adapter.py

```diff
@@ -1442,6 +1443,59 @@ def test__get_relation_type_iceberg(self, dbt_debug_caplog, mock_aws_service):
     def test__is_current_column(self, column, expected):
         assert self.adapter._is_current_column(column) == expected
 
+    def test_format_partition_keys(self):
+        partition_keys = ["year(date_col)", "bucket(col_name, 10)", "default_partition_key"]
+        expected_output = "date_trunc('year', date_col), col_name, default_partition_key"
+        assert self.adapter.format_partition_keys(partition_keys) == expected_output
+
+    def test_format_one_partition_key(self):
+        assert self.adapter.format_one_partition_key("month(hidden)") == "date_trunc('month', hidden)"
+        assert self.adapter.format_one_partition_key("bucket(bucket_col, 10)") == "bucket_col"
+        assert self.adapter.format_one_partition_key("regular_col") == "regular_col"
+
+    def test_murmur3_hash_with_int(self):
+        bucket_number = self.adapter.murmur3_hash(123, 100)
+        assert isinstance(bucket_number, int)
+        assert 0 <= bucket_number < 100
+        assert bucket_number == 54
+
+    def test_murmur3_hash_with_datetime(self):
+        dt = datetime.datetime.now()
+        bucket_number = self.adapter.murmur3_hash(dt, 100)
+        assert isinstance(bucket_number, int)
+        assert 0 <= bucket_number < 100
+
+    def test_murmur3_hash_with_str(self):
+        bucket_number = self.adapter.murmur3_hash("test_string", 100)
+        assert isinstance(bucket_number, int)
+        assert 0 <= bucket_number < 100
+        assert bucket_number == 88
+
+    def test_murmur3_hash_uniqueness(self):
+        # Ensuring different inputs produce different hashes
+        hash1 = self.adapter.murmur3_hash("string1", 100)
+        hash2 = self.adapter.murmur3_hash("string2", 100)
+        assert hash1 != hash2
+
+    def test_murmur3_hash_with_unsupported_type(self):
+        with pytest.raises(TypeError):
+            self.adapter.murmur3_hash([1, 2, 3], 100)
+
+    def test_format_value_for_partition(self):
```

Comment:
we can use https://docs.pytest.org/en/7.3.x/how-to/parametrize.html here I guess

Reply:
Used parametrize here ✅